web/validators/support_author_validator.ex

```diff
@@ -0,0 +1,34 @@
+defmodule ParticipateApi.SupportAuthorValidator do
+  import Ecto.Changeset
+  import Ecto.Query
+  alias ParticipateApi.Support
+  alias ParticipateApi.Repo
+
+  def validate_no_previous_support_given_by_author(changeset) do
+    author_id = get_field(changeset, :author_id)
+
+    query = from Support, where: [author_id: ^author_id]
+
+    if Repo.one(query) do
```

Comment:
Possible problems:
* The query seems only to select by `author_id`, but should probably also select by `proposal_id`.
* `Repo.one` seems to crash as soon as there is more than one result. Getting at most one result may be a sensible assumption, if the selector is fixed (see above) and the database is consistent. OTOH,  it may be more defensive to deal also with the case of more than one result, without risking a crash.